### ISSUE TITLE
training-ner example does not get expected results

### ISSUE THREAD

**Issue by dayuruozhi** [explosion/spaCy#1080]
I am using the example animal code to train new entity type from https://github.com/explosion/spacy/blob/master/examples/training/train_new_entity_type.py  The example can't recognize the new Entity type "ANIMAL".  Does it require spacy 1.9?

The only modification is  to replace

loss += nlp.entity.update(doc, gold, drop=0.9)

with

loss += nlp.entity.update(doc, gold)

The output is:

Loading initial model en
Ents in 'Do you like horses?':
DATE you like horses
Loading from animal
Loaded from animal
DATE you like horses

## Your Environment

* **spaCy version:** 1.8.2
* **Platform:** Linux-3.10.0-514.16.1.el7.x86_64-x86_64-with-redhat-7.3-Maipo
* **Python version:** 3.6.1
* **Installed models:** en

When I tested the same code on a different machine with less memory.  The output is:

Loading initial model en
Ents in 'Do you like horses?':
LOC horses
Loading from animal
Killed

## Info about spaCy

* **Installed models:** en
* **Platform:** Linux-4.10.0-20-generic-x86_64-with-Ubuntu-17.04-zesty
* **Python version:** 3.5.3
* **spaCy version:** 1.8.2

 I also tried using train_ner()   in the example from https://spacy.io/docs/usage/training-ner. There was no different in output.

**Comment by My3vilMe** [comment]
Hello dayuruozhi,

I am having the same problem on a windows 10 pc. Did you find a solution by any chance?
I am trying myself atm but no successful change by now. 

i dropped the 
   random.seed(0)
   nlp.entity.model.learn_rate = 0.001

and i got the result:

Loading initial model method
Ents in 'What is a horse? What is London':
LOC Horses
GPE London
test end

but the second time the result was: ( I made a second sentence)

Loading initial model method
Ents in 'What are Horses? What is London':
WORK_OF_ART are Horses
GPE London
Ents in 'Horses are Horses. Horses are animals':
ANIMAL Horses
WORK_OF_ART are Horses
DATE Horses are animals
test end

so I know it knows the Entity ANIMAL but it is totallay random every time I try again.

**Comment by dayuruozhi** [comment]
I compiled the latest code cloned from git,  then the example works perfectly.
It indeed requires Spacy 1.9 

# python3 animal.py en animal
Loading initial model en
Ents in 'Do you like horses?':
ANIMAL horses
Loading from animal
Loaded from animal
ANIMAL horses

**Comment by dayuruozhi** [comment]
Later I found out that with released 1.8 code, the example worked if I added 2 more sentences for training.

**Comment by dayuruozhi** [comment]
The output is indeed unpredicatable.  I added a sentence "I like horses." in the training. After that horses can be recognized in 'Do you like horses?' 

Next step, I inserted the code after calling train_ner()  to recognize entity from sentence" I have been to France". "France" can't be recognized any longer as "GEO", which worked with original model.

Then I inserted the same code before calling train_ner(). This time  France is recognized both before and after training.  I guess every time calling nlp(), the model is learning, and the memory of "France" is rather elusive.

**Comment by My3vilMe** [comment]
Thank you for your help. It worked for me too now!

**Comment by prakhar21** [comment]
@dayuruozhi How did you manage to run the example. I am getting the same error what you got.
Instead i am running on python 2.7.12 and Spacy 1.9
